lua/lazyvim/config/options.lua

```diff
@@ -4,6 +4,13 @@ vim.g.maplocalleader = "\\"
 
 -- LazyVim auto format
 vim.g.autoformat = true
+-- LazyVim auto format mode (files | changes)
+-- if set to changes only format actual changes based on git hunks
+-- otherwise format all the file
+vim.g.formatmode = "file"
+--- LazyVim auto format changes only mode will be ignored
+--- for these filetypes and fallback to file
+vim.g.format_range_exclude_ft = { "lua" }
```

Comment:
Do these need to be set specifically, since in `/util/format.lua` you do 
```lua
local gmode = (vim.g.formatmode == nil and "file") or vim.g.formatmode
```
and 
```lua
if vim.tbl_contains(vim.g.format_range_exclude_ft or M.format_range_exclude_ft, vim.bo[buf].filetype) then
    bmode = "forced file"
end
```
respectively? Also for the second snippet maybe
```lua
vim.tbl_contains(LazyVim.dedup(vim.list_extend(vim.g.format_range_exclude_ft or {}, M.format_range_exclude_ft)), vim.bo[buf].filetype)
```
would be more appropriate, so that it neither contains duplicate filetypes nor the `vim.g.format_range_exclude_ft` overwrites the `M.format_range_exclude_ft` in case `vim.g.format_range_exclude_ft` does not contain the `lua` filetype. Whereas if you use `vim.list_extend` it will just extend the filetype patterns. That goes for both `M.register` and `M.format` functions.

Reply:
Yes, you are right about the overwritable thing. I didn't think it through. What I proposed would just ensure that the ft `lua` would always be included, which might be not desirable by some users. Feel free to mark this as resolved.